### PR TITLE
Add the general configuration

### PR DESCRIPTION
**Pull request by voscob**
### Description

I've added a general configuration and validator.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-redshift/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config/configuration_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"go.uber.org/multierr"
+)
+
+const (
+	testDSN   = "postgres://username:password@endpoint:5439/database"
```
[the remark this answers is not in the document]
Comment (voscob) [reply config/configuration_test.go:26]: I’m going to use the Postgres client. The same syntax.

[the remark this answers is not in the document]
Comment (maksenius) [reply config/configuration_test.go:26]: hm interesting
Did you check is it works fine?

[the remark this answers is not in the document]
Comment (voscob) [reply config/configuration_test.go:26]: Yep, I did. 
But if anything goes wrong, I’ll update it.

[the remark this answers is not in the document]
Comment (maksenius) [reply config/configuration_test.go:26]: ok

**Review by maksenius — APPROVED** [review]

**Pull request merged**